File: sources/java-api/src/main/java/com/example/entities/Booking.java
package com.example.entities;

import java.io.Serializable;
import java.sql.Date;

import com.example.entities.Entities;

public class Booking extends Entities implements Serializable {
    private static final long serialVersionUID = 1L;
    private int user_id;
    private int n_people;
    private Date date;
    private int hour;
    private int minute;
    private int activity_id;

    public Booking(){}

    /**
     * @return the hour
     */
    public int getHour() {
        return hour;
    }

    /**
     * @param hour the hour to set
     */
    public void setHour(int hour) {
        this.hour = hour;
    }

    /**
     * @return the minute
     */
    public int getMinute() {
        return minute;
    }

    /**
     * @param minute the minute to set
     */
    public void setMinute(int minute) {
        this.minute = minute;
    }

    /**
     * @return the activity_id
     */
    public int getActivity_id() {
        return activity_id;
    }

    /**
     * @param activity_id the activity_id to set
     */
    public void setActivity_id(int activity_id) {
        this.activity_id = activity_id;
    }

    public Booking(int id, int user_id, int n_people, Date date, int hour, int minute, int activity_id) {
        this.id = id;
        this.user_id = user_id;
        this.n_people = n_people;
        this.date = date;
        this.hour = hour;
        this.minute = minute;
        this.activity_id = activity_id;
    }

    /**
     * @return the user_id
     */
    public int getUser_id() {
        return user_id;
    }

    /**
     * @return the date
     */
    public Date getDate() {
        return date;
    }

    /**
     * @param date the date to set
     */
    public void setDate(Date date) {
        this.date = date;
    }

    /**
     * @return the n_people
     */
    public int getN_people() {
        return n_people;
    }

    /**
     * @param n_people the n_people to set
     */
    public void setN_people(int n_people) {
        this.n_people = n_people;
    }

    /**
     * @param user_id the user_id to set
     */
    public void setUser_id(int user_id) {
        this.user_id = user_id;
    }
}

File: sources/web/modele/entities/Booking.php
<?php

require_once('Entities.php');

/**
* 
*/
class Booking extends Entities implements JsonSerializable
{

	private $_user_id;
    private $_n_people;
    private $_date;
    private $_hour;
    private $_minute;
	private $_activity_id;
	
	public function __construct()
	{

	}

	public static function create($stdClass)
    {
        if ($stdClass == null) {
            return null;
        }
        $booking = new Booking();
        $booking->setUser_Id($stdClass->user_id);
        $booking->setN_People($stdClass->n_people);  
        $booking->setDate($stdClass->date);
        $booking->setHour($stdClass->hour);
        $booking->setMinute($stdClass->minute);      
        $booking->setActivity_Id($stdClass->activity_id);
        return $booking;
    }
    public function getDate(){
        return $this->_date;    
    }

    public function setDate($date){
        $this->_date = $date;
    }

    public function getN_People()
    {
        return $this->_n_people;
    }

    public function getUser_Id()
    {
        return $this->_user_id;
    }

    public function getActivity_Id()
    {
        return $this->_activity_id;
    }

    public function setN_People($n_people)
    {
        $this->_n_people = $n_people;
    }

    public function setUser_Id($id_user)
    {
        $this->_user_id = $id_user;
    }

    public function setActivity_Id($id_hobbyactivity)
    {
        $this->_activity_id = $id_hobbyactivity;
    }

    public function toArray()
    {
        $array = array(
            'id' => $this->_id,
            'user_id' => $this->_user_id,
            'n_people' => $this->_n_people,
            'date' => $this->date,
            'hour' => $this->hour,
            'minute' => $this->minute,
            'activity_id' => $this->_activity_id
        );
        return $array;
    }

    public function toString()
    {
        return __class__;
    }

    public function jsonSerialize() 
    {
        return $this->toArray();
    }





    /**
     * Get the value of _hour
     */ 
    public function getHour()
    {
        return $this->_hour;
    }

    /**
     * Set the value of _hour
     *
     * @return  self
     */ 
    public function setHour($_hour)
    {
        $this->_hour = $_hour;

        return $this;
    }

    /**
     * Get the value of _minute
     */ 
    public function getMinute()
    {
        return $this->_minute;
    }

    /**
     * Set the value of _minute
     *
     * @return  self
     */ 
    public function setMinute($_minute)
    {
        $this->_minute = $_minute;

        return $this;
    }
}



?>